xcb.c

```diff
@@ -199,10 +202,18 @@ void grab_pointer_and_keyboard(xcb_connection_t *conn, xcb_screen_t *screen, xcb
         /* Make this quite a bit slower */
         usleep(50);
 
-        /* Measure elapsed time and trigger a screen redraw if elapsed > 250000 */
+        struct timeval now;
+        if (gettimeofday(&now, NULL) == -1) {
+            err(EXIT_FAILURE, "gettimeofday");
+        }
+
+        struct timeval elapsed;
+        timersub(&now, &start, &elapsed);
+
+        /* Trigger a screen redraw if 100ms elapsed */
         if (!redrawn &&
             (tries % 100) == 0 &&
-            (clock() - start) > 250000) {
+            elapsed.tv_usec > 100000) {
```

Comment:
Is there a specific reason to change the timing itself? Also, why don't we consider using much, much fewer grabbing retries to cut down the time? xscreensaver only uses a handful. This would give much quicker feedback.

Reply:
See the PR description:

> Changes the magic number to 100ms, which is commonly believed to be the threshold at which humans perceive changes as “instantaneous”. This is large enough to prevent flickering, and small enough to fix the issue.

We can also cut down the number of retries, but that seems orthogonal: the number of retries determines how long i3lock can hang, not how quickly the user is presented with feedback — that would still be coupled to wall-clock time measurement, as retries are not deterministic in length either (just like `clock`, which we’re removing here).